nearpy/storage/storage_memory.py

```diff
@@ -49,6 +49,14 @@ def store_vector(self, hash_name, bucket_key, v, data):
             self.buckets[hash_name][bucket_key] = []
         self.buckets[hash_name][bucket_key].append((v, data))
 
+    def delete_vector(self, hash_name, data):
+        """
+        Deletes vector and JSON-serializable data in bucket with specified key.
+        """
+        for bucket_key in self.buckets[hash_name]:
+            self.buckets[hash_name][bucket_key] = \
+                [(v, id_data) for v, id_data in self.buckets[hash_name][bucket_key] if id_data != data]
```

Comment:
@amorgun did you do it by your own or is there a tool that factors it automatically? Would you recommend me resources (books, videos...) for good practices? (I mean, testing, PEP8...)